scripts/lunch.coffee

```diff
@@ -28,11 +28,94 @@ location= '45.257217,19.845845'
 https = require 'https'
 
 module.exports = (robot) ->
+
+  robot.brain.data.lunch = {}
+
+  is_brain_empty = () ->
+    for x,y of robot.brain.data.lunch
+      return false
+    return true
+
+  room = (msg) ->
+    msg.message.user.room
+
+  location_exists = (location) ->
+    location of robot.brain.data.lunch
+
+  lunch_list = () ->
+    console.log robot.brain.data.lunch
+    for location, users of robot.brain.data.lunch
+      users_for_location = ""
+      for user, i in users
+        users_for_location += user.name
+        users_for_location += "\n"
+      title: location
+      value: users_for_location
+      short: false
+
+  add_user_on_location = (msg) ->
+    location = msg.match[1]
+    if(!location_exists(location))
```

Comment:
The biggest reason is readability. You can maybe take a look at this too https://github.com/bbatsov/ruby-style-guide. (I know it is ruby specific, but many things apply to coffescript as well).